### PR TITLE
[feat] Add P series example nodegroup to jark stack example

### LINKED ISSUE #370: GPU node group (P4d/P5)  configuration with EFA, CBR and RAID0 for SSD
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### What is the outcome that you are trying to reach?

- Show how to create a new node group with 2 GPU instances (P4d)  by configuring 8 EFA devices, Capacity Block reservations(CBR) and Placement groups 
- Also, configure 8TB SSD offered by p4d using bootstrap commands and configure RAID0

### PR DESCRIPTION
**Pull request by alanty**
### What does this PR do?
Adds a nodegroup definition for a P5.48xlarge group, with EFA devices, a Capacity Reservation, and RAID0 over the NVMe devices. That nodegroup can be used fora P4 instance with minor changes noted in the comments. 
Closes #370 

This also adds the EFA plugin from the data addons, this should be usable on any instance with EFA devices not just the P series.

### Motivation

Add an example/reference for the P series with the plugins and config for use in the jark stack.

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Mandatory for new blueprints. Yes, I have added a example to support my blueprint PR
- [ ] Mandatory for new blueprints. Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### Additional Notes

The example in this repo looks to be based on CUDA11 and Nvidia 470 drivers. These look to have issues with the P5, CUDA12 and the Nvidia 535 drivers. I've got a rough update in progress, we'll work on a different PR for that change.

### CODE REVIEW

File: ai-ml/jark-stack/terraform/eks.tf [hunk 1]
```diff
@@ -147,5 +147,66 @@ module "eks" {
         Name = "gpu-node-grp"
       })
     }
+
+    # # This nodegroup can be used for P4/P5 instances with, or without, a Capacity Reservation.
+    # #
```
Comment (lusoal) [inline ai-ml/jark-stack/terraform/eks.tf:152]: @alanty is this comment required? We need to make sure that we have the instructions when to remove the comments of this section, might want to add into the website folder documentation for the JARK stack

**Review by lusoal — APPROVED** [review]
LGTM, just need clarification around how we should deal with the commented section

File: ai-ml/jark-stack/terraform/eks.tf (hunk 1 above)
Comment (vara-bonthu) [reply ai-ml/jark-stack/terraform/eks.tf:152]: @lusoal, the JARK stack currently missing documentation for the DoEKS website. I agree that adding a section to explain how to enable and deploy the P5 nodegroup with EFA and CBR would be beneficial. We can do that as a part JARK deployment Website doc.

**Review by vara-bonthu — APPROVED** [review]

**Review by vara-bonthu — APPROVED** [review]
LGTM

**Pull request merged**